include/Spectra/LinAlg/Orthogonalization.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2020 Netherlands eScience Center <f.zapata@esciencecenter.nl>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef SPECTRA_ORTHOGONALIZATION_H
+#define SPECTRA_ORTHOGONALIZATION_H
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+namespace Spectra {
+
+template <typename Matrix>
+Eigen::Index sanity_check(Matrix& in_output, Eigen::Index leftColsToSkip = 0)
+{
+    assert(in_output.cols() > leftColsToSkip && "leftColsToSkip is larger than columns of matrix");
+    assert(leftColsToSkip >= 0 && "leftColsToSkip is negative");
+    if (leftColsToSkip == 0)
+    {
+        in_output.col(0).normalize();
+        leftColsToSkip = 1;
+    }
+    return leftColsToSkip;
+}
```

Comment:
I do not like the name, because it is more than a check.

Reply:
true ! ill fix it after lunch